File: app/presentation/assets/js/shop.js
$(document).ready(function(){
  // $('#toast').toast('show');

  $('#listrow').hide();
  $('#menu_panel').hide();
  $('#review_panel').hide();

  $('#listbtn').click(function(){
    $('#maprow').hide();
    $('#listrow').show();
    $('#mapbtn').removeClass('active');
    $('#listbtn').addClass('active');
  });

  $('#mapbtn').click(function(){
    $('#maprow').show();
    $('#listrow').hide();
    $('#listbtn').removeClass('active');
    $('#mapbtn').addClass('active');
  });

  $('#menubtn').click(function(){
    $('#menu_panel').show("slide", {
      direction: "right"
    }, 500);
  })

  $('#reviewbtn').click(function(){
    $('#review_panel').show("slide", {
      direction: "right"
    }, 500);
  })

  $('#menubackbtn').click(function(){
    $('#menu_panel').hide("slide", {
      direction: "right"
    }, 500);
  })

  $('#reviewbackbtn').click(function(){
    $('#review_panel').hide("slide", {
      direction: "right"
    }, 500);
  })
});

function make_toast_info(shop){
  $('#toast .name').text(shop.name);
  $('#toast .address').text(shop.address);
  $('#toast .phone_number').text(shop.phone_number);
  $('#toast .opening_now').text(shop.opening_now);
  $('#toast .rating').text(shop.rating);
  $('#toast .recommend_drink').text(shop.recommend_drink);

  make_review_info(shop.reviews);
  make_menu_info(shop.menu);
}

function make_review_info(reviews){
  $('#review_info').text("");

  reviews.forEach((review, i) => {
    author = '<div>' + review.author + ' - ' + review.relative_time + '</div>';
    rating = '<div>' + "&#9733".repeat(review.rating) + '</div>'
    content ='<div>' + review.content + '</div>';
    output = author + rating + content + '</br>';
    $('#review_info').append(output);
  });
}

function make_menu_info(menu){
  $('#menu_info').text("");

  menu.forEach((drink, i) => {
    var output = '<tr>' +
                 '<td>' + drink.name + '</td>' +
                 '<td>' + drink.category + '</td>' +
                 '<td>' + drink.price.M + '</td>' +
                 '<td>' + drink.price.L + '</td>' +
                 '</tr>';

    $('#menu_info').append(output);
  });
}

File: app/presentation/view_objects/shop.rb
# frozen_string_literal: true

module Views
  # View for shop and shop extractions(recommend drink, menu) for a given shop
  class Shop
    def initialize(shop, recommend_drink = nil, menu = nil)
      @shop = shop
      @recommend_drink = recommend_drink
      @menu = menu
    end

    def entity
      @shop
    end

    def name
      @shop.name
    end

    def address
      @shop.address
    end

    def latitude
      @shop.latitude
    end

    def longitude
      @shop.longitude
    end

    def phone_number
      @shop.phone_number
    end

    def map_url
      @shop.map_url
    end

    def opening_now
      @shop.opening_now
    end

    def rating
      @shop.rating
    end

    def reviews
      @shop.reviews.map { |review| Review.new(review) }
    end

    def recommend_drink
      @recommend_drink
    end

    def menu
      @menu
    end

    def json_format_to_js
      format_shop = @shop.to_h
      format_shop[:recommend_drink] = @recommend_drink
      format_shop[:menu] = @menu
      format_shop.to_json
    end
  end
end
